example_configs/tomcat.yml

```diff
@@ -1,33 +1,409 @@
+  name: tomcat_stringcache_access_total
+  help: 'Tomcat: The number of times the string cache was accessed.'
+  type: COUNTER
+- pattern: '^Tomcat<type=StringCache><>cacheSize:'
+  name: tomcat_stringcache_size
+  valueFactor: 0.001
+  help: 'Tomcat: The number of entries in the string cache.'
+  type: GAUGE
+- pattern: '^Tomcat<type=StringCache><>hitCount:'
+  name: tomcat_stringcache_hits_total
+  help: 'Tomcat: The number of times a String was obtained from the cache for a ByteChunk or CharChunk'
+  type: COUNTER
+- pattern: '^Tomcat<type=StringCache><>trainThreshold:'
+  name: tomcat_stringcache_train_threshold
+  help: 'Tomcat: The number of times toString() must be called before the string cache is activated.'
+  type: GAUGE
+
+# ------------------------ ThreadPool -----------------------
+
+- pattern: '^Tomcat<type=ThreadPool, name="(.*)"><>currentThreadCount:'
+  name: tomcat_threadpool_threads
+  help: 'Tomcat: The number of threads that are managed by the pool.'
+  type: GAUGE
+  labels:
+    pool: $1
+- pattern: '^Tomcat<type=ThreadPool, name="(.*)"><>currentThreadsBusy:'
+  name: tomcat_threadpool_threads_busy
+  help: 'Tomcat: The number of threads that are in use to handle connections.'
+  type: GAUGE
+  labels:
+    pool: $1
+- pattern: '^Tomcat<type=ThreadPool, name="(.*)"><>connectionCount:'
+  name: tomcat_threadpool_connections
+  help: 'Tomcat: Current count of connections handled by this endpoint, if the connections are counted (which happens when the maximum count of connections is limited), or -1 if they are not.'
+  type: GAUGE
+  labels:
+    pool: $1
+
+# Let's just dump all these for now, without the metric type or units suffix:
+# - acceptorThreadCount
+# - acceptorThreadPriority
+# - backlog
+# - executorTerminationTimeoutMillis
+# - keepAliveCount
+# - keepAliveTimeout
+# - localPort
+# - maxConnections
+# - maxHeaderCount
+# - maxKeepAliveRequests
+# - maxThreads
+# - minSpareThreads
+# - oomParachute
+# - pollerThreadCount
+# - pollerThreadPriority
+# - port
+# - selectorTimeout
+# - soLinger
+# - soTimeout
+# - threadPriority
+- pattern: '^Tomcat<type=ThreadPool, name="(.*)"><>(.*):'
+  name: tomcat_threadpool_$2
+  labels:
+    pool: $1
+
+# ------------------------ WebModule -----------------------
+# These already have a help text, so we let it pass through.
+
+# Cumulative error count of all servlets in this context (Tomcat<j2eeType=WebModule, name=//localhost/, J2EEApplication=none, J2EEServer=none><>errorCount)
+# Cumulative request count of all servlets in this context (Tomcat<j2eeType=WebModule, name=//localhost/, J2EEApplication=none, J2EEServer=none><>requestCount)
+- pattern: '^Tomcat<j2eeType=WebModule, name=//localhost(.*), J2EEApplication=none, J2EEServer=none><>requestCount:'
+  name: tomcat_webmodule_requests_total
+  type: COUNTER
+  labels:
+    context: $1
+- pattern: '^Tomcat<j2eeType=WebModule, name=//localhost(.*), J2EEApplication=none, J2EEServer=none><>errorCount:'
```

Comment:
This way of extracting context is different to the ones above. Will it always have the same content and meaning?